src/main/java/com/autotune/common/datasource/DataSourceCollection.java

```diff
@@ -0,0 +1,222 @@
+                LOGGER.info(KruizeConstants.DataSourceConstants.DataSourceInfoMsgs.VERIFYING_DATASOURCE_REACHABILITY + name);
+                DataSourceOperatorImpl op = DataSourceOperatorImpl.getInstance().getOperator(KruizeConstants.SupportedDatasources.PROMETHEUS);
+                if (op.isServiceable(url) == CommonUtils.DatasourceReachabilityStatus.REACHABLE) {
+                    LOGGER.info(KruizeConstants.DataSourceConstants.DataSourceSuccessMsgs.DATASOURCE_SERVICEABLE);
+                    dataSourceCollection.put(name, datasource);
+                    LOGGER.info(KruizeConstants.DataSourceConstants.DataSourceSuccessMsgs.DATASOURCE_ADDED);
+                } else {
+                    throw new DataSourceNotServiceable(KruizeConstants.DataSourceConstants.DataSourceErrorMsgs.DATASOURCE_NOT_SERVICEABLE);
+                }
+            } else {
+                throw new UnsupportedDataSourceProvider(KruizeConstants.DataSourceConstants.DataSourceErrorMsgs.UNSUPPORTED_DATASOURCE_PROVIDER);
+            }
+        } catch (UnsupportedDataSourceProvider e) {
+            LOGGER.error(e.getMessage());
+        } catch (DataSourceNotServiceable e) {
+            LOGGER.error(e.getMessage());
+        } catch (DataSourceAlreadyExist e) {
+            LOGGER.error(e.getMessage());
+        }
+    }
+
+    /**
+     * Loads the data sources available at installation time
+     * @param configFileName name of the config file mounted
+     * TODO:// add support for datasource operations
+     */
+    public void addDataSourcesFromConfigFile(String configFileName) {
+        try {
+            String configFile = System.getenv(configFileName);
+            JSONObject configObject = null;
+            // ValidationOutputData addedToDB = null;
+
+            InputStream is = new FileInputStream(configFile);
+            String jsonTxt = new String(is.readAllBytes(), StandardCharsets.UTF_8);
+            configObject = new JSONObject(jsonTxt);
+            JSONArray dataSourceArr = configObject.getJSONArray(KruizeConstants.DataSourceConstants.KRUIZE_DATASOURCE);
+
+            for (Object dataSourceObj: dataSourceArr) {
+                JSONObject dataSourceObject = (JSONObject) dataSourceObj;
+                String name = dataSourceObject.getString(KruizeConstants.DataSourceConstants.DATASOURCE_NAME);
+                // check the DB if the datasource already exists
+//                try {
+//                    DataSourceInfo dataSourceInfo = new ExperimentDBService().loadDataSourceFromDBByName(name);
+//                    if (null != dataSourceInfo) {
+//                        LOGGER.error("Datasource: {} already exists!", name);
+//                        continue;
+//                    }
+//                } catch (Exception e) {
+//                    LOGGER.error("Loading saved datasource {} failed: {} ", name, e.getMessage());
+//                }
```

Comment:
In general we do not merge commented code, please remove

Reply:
Updating the commented code for db operations according to recently merged db tables.